SparseGrids/gridtestExternalTests.cpp

```diff
@@ -1041,37 +1052,81 @@ bool ExternalTester::testAllPWLocal() const{
     int wfirst = 10, wsecond = 35, wthird = 15;
 
     { TasGrid::TypeOneDRule oned = TasGrid::rule_semilocalp;
-    const int depths1[18] = { 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8 };
-    const double tols1[18] = { 1.E-03, 5.E-01, 5.E-01, 1.E-03, 1.E-03, 1.E-03, 1.E-07, 1.E-04, 1.E-04, 1.E-07, 1.E-05, 1.E-05, 1.E-07, 4.E-06, 4.E-06, 1.E-07, 4.E-06, 4.E-06 };
-    if (testLocalPolynomialRule(&f21nx2, oned, depths1, tols1)){
-        if (verbose) cout << setw(wfirst) << "Rule" << setw(wsecond) << IO::getRuleString(oned) << setw(wthird) << "Pass" << endl;
-    }else{
-        cout << setw(wfirst) << "Rule" << setw(wsecond) << IO::getRuleString(oned) << setw(wthird) << "FAIL" << endl; pass = false;
-    }}
+      const int depths1[30] = { 8, 8, 8, 8, 8,
+                                8, 8, 8, 8, 8,
+                                8, 8, 8, 8, 8,
+                                8, 8, 8, 8, 8,
+                                8, 8, 8, 8, 8,
+                                8, 8, 8, 8, 8};
+      const double tols1[30] = { 1.E-03, 5.E-01, 5.E-01, 5.E-00, 5.E-00,
```

Comment:
I changed the tolerance for the zeroth order test to `5E-01`, but this actually doesn't matter because I have a special case that only checks if the return derivative is zero (i.e, the tolerance does not impact this test).

I also added comments to each row of the tolerances, as requested.